simtools/simulator.py

```diff
@@ -0,0 +1,740 @@
+import logging
+import re
+from collections import defaultdict
+from copy import copy
+from pathlib import Path
+
+import astropy.units as u
+import numpy as np
+
+import simtools.config as cfg
+import simtools.io_handler as io
+import simtools.util.general as gen
+from simtools.job_submission.job_manager import JobManager
+from simtools.model.array_model import ArrayModel
+from simtools.simtel.simtel_histograms import SimtelHistograms
+from simtools.corsika.corsika_runner import CorsikaRunner
+from simtools.simtel.simtel_runner_array import SimtelRunnerArray
+from simtools.util import names
+
+__all__ = ["Simulator"]
+
+
+class InvalidRunsToSimulate(Exception):
+    pass
+
+
+class Simulator:
+    """
+    Simulator is responsible for managing simulation of showers and array of telescopes. \
+    It interfaces with simulation software-specific packages, like CORSIKA or sim_telarray.
+
+    The configuration is set as a dict configData or a yaml \
+    file configFile.
+
+    Example of configData for shower simulations:
+
+    .. code-block:: python
+
+        self.configData = {
+            'dataDirectory': '.',
+            'site': 'South',
+            'layoutName': 'Prod5',
+            'runRange': [1, 100],
+            'nshow': 10,
+            'primary': 'gamma',
+            'erange': [100 * u.GeV, 1 * u.TeV],
+            'eslope': -2,
+            'zenith': 20 * u.deg,
+            'azimuth': 0 * u.deg,
+            'viewcone': 0 * u.deg,
+            'cscat': [10, 1500 * u.m, 0]
+        }
+
+
+    Example of configData for array simulations:
+
+    .. code-block:: python
+
+    configData = {
+        'dataDirectory': '(..)/data',
+        'primary': 'gamma',
+        'zenith': 20 * u.deg,
+        'azimuth': 0 * u.deg,
+        'viewcone': 0 * u.deg,
+        # ArrayModel
+        'site': 'North',
+        'layoutName': '1LST',
```

Comment:
Is this really a layout name? Or is it a name the user provides and isn't checked?

Reply:
This is a layout as defined in data/layout/